python-package/xgboost/core.py

```diff
@@ -1234,17 +1234,33 @@ def _set_data_from_csr(self, csr: scipy.sparse.csr_matrix) -> None:
         )
 
 
-class DeviceQuantileDMatrix(DMatrix):
-    """Device memory Data Matrix used in XGBoost for training with tree_method='gpu_hist'. Do
-    not use this for test/validation tasks as some information may be lost in
-    quantisation. This DMatrix is primarily designed to save memory in training from
-    device memory inputs by avoiding intermediate storage. Set max_bin to control the
-    number of bins during quantisation.  See doc string in :py:obj:`xgboost.DMatrix` for
-    documents on meta info.
+class QuantileDMatrix(DMatrix):
+    """A DMatrix variant that generates quantilized data directly from input for
+    ``hist`` and ``gpu_hist`` tree methods. This DMatrix is primarily designed to save
+    memory in training by avoiding intermediate storage. Set ``max_bin`` to control the
+    number of bins during quantisation, which should be consistent with the training
+    parameter ``max_bin``. When ``QuantileDMatrix`` is used for validation/test dataset,
```

Comment:
Would be good to have slightly more information here.
e.g.
"Supplying the training DMatrix as a reference means that the same quantisation applied to the training data is applied to the validation/test data."

Reply:
Thank you for the suggestion. Added as part of the document for `ref`.